src/lib.rs

```diff
@@ -29,7 +29,7 @@ mod util;
 
 #[doc(include = "../docs/notes.md")]
 mod notes {}
-mod circuit_proof;
+mod constraint_system;
```

Comment:
(Maybe in the separate PR or even later) My bikeshed preference for names is: 

1. Rename private module `constraint_system` into `cs`.
2. Rename public module with reexports from `r1cs` into `cs`. All crate-private identifiers could be made `pub(crate)` or we can do `use cs as cs_internal` inside `mod.rs` so we avoid using `pub(crate)` and can create another module `cs` for public re-exports.
3. Rename `R1CSProof` into `ConstraintSystemProof`.

Reply:
I'm also down with `s/cs/constraint_system/` if you prefer the spelled out name. But for both private and public modules.

Reply:
How about internal “cs” and external “constraint_system”?

Reply:
I think that "R1CS" is a good name and that we should keep it.